apps/src/applab/dataBrowser/EditTableRow.jsx

```diff
@@ -1,35 +1,100 @@
+import FirebaseStorage from '../firebaseStorage';
 import Radium from 'radium';
 import React from 'react';
+import { castValue, displayValue } from './dataUtils';
 import * as dataStyles from './dataStyles';
 
 const EditTableRow = React.createClass({
   propTypes: {
     columnNames: React.PropTypes.array.isRequired,
+    tableName: React.PropTypes.string.isRequired,
     record: React.PropTypes.object.isRequired
   },
 
+  getInitialState() {
+    return {
+      isEditing: false
```

Comment:
would be good to initialize newRecord here, just so that one can easily see what the state is. I guess the fact we need a nested object here (so that we dont have to worry about conflicting with isEditing) is an argument in favor of doing the same in the AddTableRow case